examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,300 @@
+        # Get global settings
+        global_settings = get_global_settings(nocache=True)
+        langroid.utils.configuration.set_global(global_settings)
+
+        # Get base LLM configuration
+        agent_config: AgentConfig = get_base_llm_config()
+        system_messages: SystemMessages = load_system_messages(
+            "system_messages.json"
+        )
+        llm_delegate: bool = is_llm_delegate()
+
+        # Select topic and sides
+        selected_topic_tuple: Optional[tuple] = select_debate_topic(system_messages)
+        if not selected_topic_tuple:
+            logger.error("No topic selected. Exiting.")
+            return
+        topic_name, pro_key, con_key = selected_topic_tuple
+        side: str = select_side(topic_name)
+
+        # Prompt for the number of debate turns
+        max_turns: int = int(
+            Prompt.ask(
+                "How many turns should the debate continue for?",
+                default="4",
+            )
+        )
+        # Create agents for pro, con, and feedback
+        pro_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages[pro_key].message
+            )
+        )
+
+        con_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages[con_key].message
+            )
+        )
+
+        feedback_agent = lr.ChatAgent(
+            lr.ChatAgentConfig(
+                llm=agent_config,
+                system_message=system_messages.messages["feedback"].message
+            )
+        )
+
+        logger.info("Pro, Con, and feedback agents Created.")
+
+        # Determine user's side
+        if side == "pro":
+            user_agent, ai_agent = pro_agent, con_agent
+            user_side, ai_side = "Pro", "Con"
+        else:
+            user_agent, ai_agent = con_agent, pro_agent
+            user_side, ai_side = "Con", "Pro"
+
+        logger.info(
+            f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+            f"LLM Delegate: {llm_delegate}"
+        )
+
+        is_user_turn: bool = True
+
+        if side == "pro":
+            agent_name = pro_agent
+            pro_agent.clear_history()
+            logger.info(f"\n{side} Agent ({topic_name}):\n")
+            if llm_delegate:
+                logger.info("Autonomous Debate Selected")
+            else:
+                user_input = Prompt.ask(
+                    "Your argument (or type 'f' for feedback, 'done' to end):"
+                )
+                pro_agent.llm_response(user_input)
+            pro_agent.enable_message(DoneTool)
+            pro_agent_task = lr.Task(pro_agent, interactive=False)
+            con_agent_task = lr.Task(con_agent, interactive=False, single_round=True)
+            pro_agent_task.add_sub_task(con_agent_task)
+            pro_agent_task.run(turns=max_turns)
```

Comment:
`task.run()` with no starting msg may not work with some models (e.g. gemini) which throw an err when the first msg after the sys msg is not a user-role msg. So you can pass in a dummy msg like `task.run("get started", turns=...)` as I did in my example

Reply:
gemini is working now. thanks for this!